src/network/netdev/batadv.c

```diff
@@ -0,0 +1,205 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <inttypes.h>
+
+#include "netdev/batadv.h"
+#include "network-internal.h"
+#include "networkd-manager.h"
+#include "netlink-util.h"
+#include <linux/genetlink.h>
+
+#include "fileio.h"
+#include "parse-util.h"
+#include "stdio-util.h"
+#include "string-util.h"
+
+static void batadv_init(NetDev *n) {
+        _unused_ Batadv *b;
+
+        b = BATADV(n);
+
+        /* Set defaults */
+        b->aggregation=true;
+        b->bridge_loop_avoidance=true;
+        b->distributed_arp_table=true;
+        b->fragmentation=true;
+        b->hop_penalty=true;
+}
+
+int config_parse_batadv_gateway_mode(
```

Comment:
Please introduce string table and use DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING(). Then, you can simply define conf parser by using DEFINE_CONFIG_PARSE_ENUM() or DEFINE_CONFIG_PARSE_ENUM_WITH_DEFAULT().